pkg/controller/jobs/statefulset/statefulset_webhook.go

```diff
@@ -62,9 +66,19 @@ func (wh *Webhook) Default(ctx context.Context, obj runtime.Object) error {
 	log := ctrl.LoggerFrom(ctx).WithName("statefulset-webhook")
 	log.V(5).Info("Applying defaults")
 
-	cqLabel, ok := ss.Labels[constants.QueueLabel]
-	if !ok {
-		return nil
+	labels := ss.Labels
+	cqLabel, ok := labels[constants.QueueLabel]
+	if !ok || cqLabel == "" {
+		if features.Enabled(features.LocalQueueDefaulting) && wh.queues.DefaultLocalQueue(ss.ObjectMeta.Namespace) != "" {
+			cqLabel = kueue_constants.DefaultLocalQueueName
+			if labels == nil {
+				labels = make(map[string]string)
+			}
+			labels[constants.QueueLabel] = cqLabel
+			ss.SetLabels(labels)
+		} else {
+			return nil
+		}
```

Comment:
I would like to reuse the `ApplyDefaultLocalQueue`. Any issue with that?

Reply:
I had an error: `*StatefulSet does not implement jobframework.GenericJob (missing method Finished)`
Same with deployment. Maybe I can easily implement those methods? 
Actually, I see those jobs doesn't apply ApplyDefaultForSuspend and doesn't rely on manageJobsWithoutQueueName. So I'm not 100% sure if LocalQueue defaulting should be applied here.

Reply:
Not sure that it is good idea, due to deployment and sts is serving workload. 

I would propose to optimize ApplyDefaultLocalQueue. Can we use client.Object instead of Generic Job?

Reply:
>Maybe I can easily implement those methods?

I meant to have no-op Finished method there. 
The main question is still open: Should the default LQ be applied there having that ApplyDefaultForSuspend and manageJobsWithoutQueueName are not applied?

Reply:
Updated function to use clientObject instead of GenericJob :)

Reply:
Nice! Thank you so much!